Sources/TestingMacros/Support/SHA256.swift

```diff
@@ -0,0 +1,184 @@
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014–2025 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+//
+
+/// The contents of this file were copied more-or-less verbatim from
+/// [swift-tools-support-core](https://github.com/swiftlang/swift-tools-support-core/blob/main/Sources/TSCBasic/HashAlgorithms.swift).
+
```

Comment:
Consider also adding the commit hash to help future patches from upstream, etc.

Reply:
We do not expect SHA-256 to change in the future, and the maintainers of TSC aren't planning to make any further changes to the original copy of this code. :)

Reply:
Added.